components/needs_team/needs_team.tsx

```diff
@@ -1,64 +1,81 @@
 const WAKEUP_CHECK_INTERVAL = 30000; // 30 seconds
 const WAKEUP_THRESHOLD = 60000; // 60 seconds
 const UNREAD_CHECK_TIME_MILLISECONDS = 10000;
 
-export default class NeedsTeam extends React.Component {
-    static propTypes = {
-        params: PropTypes.object,
-        currentUser: PropTypes.object,
-        currentChannelId: PropTypes.string,
-        currentTeamId: PropTypes.string,
-        actions: PropTypes.shape({
-            fetchMyChannelsAndMembers: PropTypes.func.isRequired,
-            getMyTeamUnreads: PropTypes.func.isRequired,
-            viewChannel: PropTypes.func.isRequired,
-            markChannelAsReadOnFocus: PropTypes.func.isRequired,
-            getTeamByName: PropTypes.func.isRequired,
-            addUserToTeam: PropTypes.func.isRequired,
-            selectTeam: PropTypes.func.isRequired,
-            setPreviousTeamId: PropTypes.func.isRequired,
-            loadStatusesForChannelAndSidebar: PropTypes.func.isRequired,
-            loadProfilesForDirect: PropTypes.func.isRequired,
-        }).isRequired,
-        theme: PropTypes.object.isRequired,
-        mfaRequired: PropTypes.bool.isRequired,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string.isRequired,
-            }).isRequired,
-        }).isRequired,
-        history: PropTypes.object.isRequired,
-        teamsList: PropTypes.arrayOf(PropTypes.object),
+declare global {
+    interface Window {
+        isActive: boolean;
+    }
+}
+
+type Props = {
+    currentUser: {
+        id: string;
+    };
+    currentChannelId?: string;
+    currentTeamId?: string;
+    actions: {
+        fetchMyChannelsAndMembers: (teamId: string) => Promise<{data: {channels: Channel[]; members: ChannelMembership[]}}>;
+        getMyTeamUnreads: () => Promise<{}>;
+        viewChannel: (channelId: string, prevChannelId?: string | undefined) => Promise<{data: boolean}>;
+        markChannelAsReadOnFocus: (channelId: string) => Promise<{}>;
+        getTeamByName: (teamName: string) => Promise<{data: Team}>;
+        addUserToTeam: (teamId: string, userId: string) => Promise<{data: TeamMembership; error?: any}>;
+        selectTeam: (team: Team) => Promise<{data: boolean}>;
+        setPreviousTeamId: (teamId: string) => Promise<{data: boolean}>;
+        loadStatusesForChannelAndSidebar: () => Promise<{}>;
+        loadProfilesForDirect: () => Promise<{}>;
+    };
+    mfaRequired: boolean;
+    match: {
+        params: {
+            team: string;
+        };
     };
+    history: {
+        push(path: string): void;
+    };
+    teamsList: Team[];
+    theme: any;
+}
 
-    constructor(params) {
-        super(params);
+type State = {
+    team: Team | null;
+    finishedFetchingChannels: boolean;
+    prevTeam: string;
+    teamsList: Team[];
+}
+
+export default class NeedsTeam extends React.Component<Props, State> {
+    public blurTime: number
```

Comment:
Nitpick, but semicolon at the end here?

Reply:
https://github.com/mattermost/mattermost-webapp/pull/4839/commits/6e4dd5c0e85debc5c48328b9b5a618bd6bc15a32